python/tank/templatekey.py

```diff
@@ -256,6 +273,145 @@ def _as_string(self, value):
         return value if isinstance(value, basestring) else str(value)
 
 
+class TimestampKey(TemplateKey):
+    """
+    Key whose value is a time string formatted with strftime.
+    """
+
+    def __init__(
+        self,
+        name,
+        utc_default=False,
+        format_spec="%Y-%m-%d-%H-%M-%S"
+    ):
+        """
+        Constructor
+        :param name: Name by which the key will be refered.
+        :param utc_default: If True, the default value will be an UTC time. Defaults to False.
```

Comment:
The ticket wasn't specific. If I had to only pick one however I would pick UTC as this it is a non-ambiguous time stamp, not localtime.

I like your idea of providing two different class types tough. If I had to pick one solution out of all the ones you suggested, I would pick that one.